docs/UnitTesting.md

```diff
@@ -46,3 +46,39 @@ You can then provide additional props, as needed, to your component (such as the
 
 At this point, your component should `mount` without errors and you can unit test your component.
 
+
+## Enabling reducers to ensure actions are dispatched
+
+If you component relies on a a reducer, e.g. redux-form submission, you can enable reducers using the `enableReducers` prop:
+
+```jsx harmony
+    myCustomEditView = mount(
+        <TestContext enableReducers>
+            <MyCustomEditView />
+        </TestContext>
+    );
+```
+
+This means that reducers will work as they will within the app.  For example, you can now submit a form and redux-form will cause a re-render of your component.
+
+
+## Spying on the store 'dispatch'
+
+If you are using `mapDispatch` within connected components, it is likely you will want to test that actions have been dispatched with the correct arguments.  You can return the `store` being used within the tests using a `renderProp`. 
+
+```jsx harmony
+    let dispatchSpy;
+    myCustomEditView = mount(
+        <TestContext>
+            {({ store }) => {
+                dispatchSpy = jest.spyOn(store, 'dispatch');
```

Comment:
smart. But I'm not sure we're note testing implementation details. @djhi, what do you think?

Reply:
yeah, i've tussled with this too. I've settled with this solution because it covers most bases, i.e normally we could test actions are dispatched by testing the UI updates, but some side-effects don't affect UI, or sometimes multiple things happen and only update the UI once.  Because of this, i think spying on the dispatch function and directly asserting it is called with the correct actions is the better way off ensuring the component/containers are written and integrated properly.  I've seen others test the `mapDispatch` function directly or only test the action was called, but both of these miss out important steps.

Happy to hear your thoughts though or show you the real-life use case for more clarity.